.github/workflows/hosted.yml

```diff
@@ -0,0 +1,290 @@
+        cmake --build out/Release --parallel
+
+    - name: Prepare artifacts
+      if: always()
+      run: |
+        cd ${{ github.workspace }}/..
+        tar czfp antlr_${{ matrix.os }}_${{ matrix.compiler }}.tgz antlr4
+        mv antlr_${{ matrix.os }}_${{ matrix.compiler }}.tgz ${{ github.workspace }}/.
+
+    - name: Archive artifacts
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: antlr_${{ matrix.os }}_${{ matrix.compiler }}
+        path: antlr_${{ matrix.os }}_${{ matrix.compiler }}.tgz
+
+
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [
+          macos-latest,
+          ubuntu-latest,
+          windows-latest
+        ]
+        target: [
+          cpp,
+          csharp,
+          dart,
+          go,
+          java,
+          javascript,
+          python2,
+          python3,
+          php,
+        ]
+
+    steps:
+    - name: Install dependencies (Ubuntu)
+      if: startswith(matrix.os, 'ubuntu')
+      run: |
+        sudo apt-get update -qq
+        sudo apt install -y ninja-build
+
+    - name: Install dependencies (MacOS)
+      if: startswith(matrix.os, 'macos')
+      run: brew install ninja
+
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'zulu'
+        java-version: 11
+
+    - name: Set up Maven
+      uses: stCarolas/setup-maven@v4.4
+      with:
+        maven-version: 3.5.4
+
+    - name: Add msbuild to PATH
+      if: startswith(matrix.os, 'windows') && (matrix.target == 'cpp')
+      uses: microsoft/setup-msbuild@v1.1
+
+    - name: Set up Python 2
+      if: matrix.target == 'python2'
+      uses: actions/setup-python@v4
+      with:
+        python-version: '2.x'
+        architecture: 'x64'
+
+    - name: Set up Python 3
+      if: matrix.target == 'python3'
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+        architecture: 'x64'
+
+    - name: Set up Node 14
+      if: matrix.target == 'javascript'
+      uses: actions/setup-node@v3
+      with:
+        node-version: '14'
+
+    - name: Setup Dotnet
+      if: matrix.target == 'csharp'
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: '6.0.x'
+
+    - name: Setup Dart 2.12.1
+      if: matrix.target == 'dart'
+      uses: dart-lang/setup-dart@v1.3
+      with:
+        sdk: 2.12.1
+
+    - name: Setup Go 1.13.1
+      if: matrix.target == 'go'
+      uses: actions/setup-go@v3
+      with:
+        go-version: '^1.13.1'
+
+    - name: Setup PHP 8.2
+      if: matrix.target == 'php'
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '8.2'
+        extensions: mbstring
+        tools: composer
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Checkout antlr PHP runtime
+      uses: actions/checkout@v2
+      with:
+        repository: antlr/antlr-php-runtime
```

Comment:
Probably the following condition:

```
if: matrix.target == 'php'
```

is missing. PHP runtime is only actual for PHP tests. It looks like the step is being executed for all runtimes: https://github.com/antlr/antlr4/runs/7371687695?check_suite_focus=true

Reply:
Fixed in  #3797